src/twisted/internet/test/test_tcp.py

```diff
@@ -750,6 +765,189 @@ def whenRun():
+            else:
+                test((-8, "Servname not supported for ai_socktype"))
+
+    def test_connectPortOverflow(self):
+        """
+        When trying to connect using a port outside of the 1-65535 range the
+        error is raised as an errback.
+        """
+
+        def test(message):
+            self.assertConnectPortError(65536, message)
+
+        if _PYPY:
+            test(("port must be 0-65535.",))
+        elif platform.isMacOSX() and self.addressClass == IPv6Address:
+            test((8, "nodename nor servname provided, or not known"))
+        elif platform.isWindows():
+            if self.reactorFactory is IOCPReactor:
+                # Windows IOCP reactor.
+                test(("value too large to convert to unsigned short",))
+            else:
+                # Windows select reactor.
+                test(("connect_ex(): port must be 0-65535.",))
+        else:
+            test(("connect_ex(): port must be 0-65535.",))
+
+    def test_connectPortZero(self):
+        """
+        An error is raised as errback when trying to listed on the
+        reserved port C{0}.
+        """
+
+        def test(message):
+            self.assertConnectPortError(0, message)
+
+        if platform.isMacOSX():
+            test(("Can't assign requested address",))
+        elif platform.isWindows():
+            if self.reactorFactory is IOCPReactor:
+                # Windows IOCP reactor.
+                test(("WSAEADDRNOTAVAIL",))
+            else:
+                # Windows select reactor.
+                test(("The requested address is not valid in its context.",))
+        else:
+            test(("Connection refused",))
+
+    def test_connectPortNotNumeric(self):
+        """
+        When trying to connect with an endpoint using a port which can not be
+        resolved to a service an errback is raised.
+        """
+        self.assertConnectPortError("invalid", ("service/proto not found ('invalid')",))
+
+    def test_getaddrinfo_negative_port(self):
+        """
+        When a negative port number is used,
+        L{socket.getaddrinfo} raises an exception on Linux and macOS,
+        while returning a result on Windows.
+
+        This is an assumption used in the implementation of L{twisted.internet.tcp._resolveIPv6}.
+        """
+
+        def resolve_negative_port():
+            """
+            Here to share the exact call between Windows and other OSes.
+            """
+            return socket.getaddrinfo("127.0.0.1", -1, 0, 0, 0, _NUMERIC_ONLY)
+
+        if platform.isWindows():
+            result = resolve_negative_port()
+            firstResolve = result[0]
+            addressResolve = firstResolve[4]
+            portResolve = addressResolve[1]
+            self.assertEqual(65535, portResolve)
+            return
+
+        with self.assertRaises(socket.gaierror):
+            resolve_negative_port()
+
+    def test_getaddrinfo_overflow_port(self):
```

Comment:
Just a coding standard nitpick.
```suggestion
    def test_getaddrinfoOverflowPort(self) -> None:
```